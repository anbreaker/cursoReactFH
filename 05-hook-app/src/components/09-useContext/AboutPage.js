import React, {useContext} from 'react';
import {UserContext} from './UserContext';

export const AboutPage = () => {
  const {user, setUser} = useContext(UserContext);

  const handleClick = () => {
    setUser({});
  };

  return (
    <div>
      <div>
        <h1>AboutPage</h1>
        <hr />
        <pre>{JSON.stringify(user, null, 3)}</pre>
        <button className="btn btn-warning" onClick={handleClick}>
          Reset
        </button>
      </div>
    </div>
  );
};
