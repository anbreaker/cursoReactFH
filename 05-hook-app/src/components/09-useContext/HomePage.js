import React, {useContext} from 'react';
import {UserContext} from './UserContext';

export const HomePage = () => {
  const {user} = useContext(UserContext);

  console.log(user);

  return (
    <div>
      <h1>HomePage</h1>
      <hr />
      <pre>{JSON.stringify(user, null, 3)}</pre>
    </div>
  );
};
