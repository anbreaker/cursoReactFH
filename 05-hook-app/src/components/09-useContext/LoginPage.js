import React, {useContext} from 'react';
import {UserContext} from './UserContext';

export const LoginPage = () => {
  const {setUser} = useContext(UserContext);

  return (
    <div>
      <div>
        <h1>LoginPage</h1>
        <hr />

        <button
          className="btn btn-primary"
          onClick={() => setUser({id: 123, name: 'anbreaker'})}>
          Login
        </button>
      </div>
    </div>
  );
};
