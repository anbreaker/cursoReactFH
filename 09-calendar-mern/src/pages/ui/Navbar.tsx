import { useDispatch, useSelector } from 'react-redux';

import { startLogout } from '../../store/actions/auth';

export const Navbar = () => {
  const { name } = useSelector((state: any) => state.auth);

  const dispatch = useDispatch();

  const handleLogout = () => {
    dispatch(startLogout());
  };

  return (
    <div className="navbar navbar-dark bg-dark mb-4">
      <span className="navbar-brand">{name}</span>

      <button className="btn btn-outline-danger" onClick={handleLogout}>
        <i className="fa fa-sign-out-alt"></i>
        <span> Exit</span>
      </button>
    </div>
  );
};
