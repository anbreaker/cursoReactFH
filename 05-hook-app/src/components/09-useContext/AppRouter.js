import React from 'react';
import {BrowserRouter as Router, Switch, Route, Redirect} from 'react-router-dom';
import {AboutPage} from './AboutPage';
import {LoginPage} from './LoginPage';
import {HomePage} from './HomePage';
import {Navbar} from './Navbar';

export const AppRouter = () => {
  return (
    <Router>
      <div>
        <Navbar />
        <div className="container">
          <Switch>
            <Route exact path="/about" component={AboutPage} />
            <Route exact path="/login" component={LoginPage} />
            <Route exact path="/" component={HomePage} />
            <Redirect to="/" />
          </Switch>
        </div>
      </div>
    </Router>
  );
};
