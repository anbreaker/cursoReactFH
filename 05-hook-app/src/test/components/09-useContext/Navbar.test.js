import {shallow} from 'enzyme';
import {Navbar} from '../../../components/09-useContext/Navbar';

describe('Pruebas en <Navbar /> ', () => {
  test('debe mostrarse correctamente', () => {
    const wrapper = shallow(<Navbar />);
    expect(wrapper).toMatchSnapshot();
  });
});
