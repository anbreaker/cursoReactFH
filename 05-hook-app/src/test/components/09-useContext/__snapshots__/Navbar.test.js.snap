// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <Navbar />  debe mostrarse correctamente 1`] = `
<nav
  className="navbar navbar-expand-sm navbar-dark bg-dark"
>
  <Link
    className="navbar-brand"
    to="/"
  >
    useContext
  </Link>
  <div
    className="collapse navbar-collapse"
    id="navbarNavAltMarkup"
  >
    <div
      className="navbar-nav"
    >
      <NavLink
        activeClassName="active"
        className="nav-item nav-link"
        exact={true}
        to="/"
      >
        Home
      </NavLink>
      <NavLink
        activeClassName="active"
        className="nav-item nav-link"
        exact={true}
        to="/about"
      >
        About
      </NavLink>
      <NavLink
        activeClassName="active"
        className="nav-item nav-link"
        exact={true}
        to="/login"
      >
        Login
      </NavLink>
    </div>
  </div>
</nav>
`;
